calculator_test.go

```diff
@@ -56,141 +49,174 @@ func TestAddRandom(t *testing.T) {
+			// function test data) but to keep test name consistent -
+			// always start with the calc function name. This way I can
+			// parse name, extract the function name and use it in the switch
+			// statement.
+			fname := strings.Split(tc.name, " ")[0]
+			var got float64
+
+			switch fname {
+			case "Add":
+				got = calculator.Add(tc.a, tc.b, tc.c...)
+			case "Multiply":
+				got = calculator.Multiply(tc.a, tc.b, tc.c...)
+			case "Subtract":
+				got = calculator.Subtract(tc.a, tc.b, tc.c...)
+			}
+
+			if got != tc.want {
+				t.Errorf("%s; %s(%f, %f, %v) = %f; want %f", tc.name, fname, tc.a, tc.b, tc.c, got, tc.want)
+			}
+		})
 	}
 }
 
-func TestMultiply(t *testing.T) {
+func TestDivide(t *testing.T) {
 	t.Parallel()
 
-	tt := []struct {
-		name        string
-		nums        []float64
-		want        float64
-		expectedErr bool
-	}{
-		{name: "Multiply three positive", nums: []float64{3, 2, 2}, want: 12, expectedErr: false},
-		{name: "Multiply two positive one negative", nums: []float64{3, 2, -2}, want: -12, expectedErr: false},
-		{name: "Multiply one positive two negative", nums: []float64{-3, 2, -2}, want: 12, expectedErr: false},
-		{name: "Multiply two positive numbers", nums: []float64{3, 2}, want: 6, expectedErr: false},
-		{name: "Multiply two negative numbers", nums: []float64{-4, -5}, want: 20, expectedErr: false},
-		{name: "Multiply one negative one positive", nums: []float64{-4, 5}, want: -20, expectedErr: false},
-		{name: "Multiply by zero", nums: []float64{-4, 0}, want: 0, expectedErr: false},
-		{name: "Multiply zero args", nums: []float64{}, want: 0, expectedErr: true},
-		{name: "Multiply one arg", nums: []float64{-4}, want: 0, expectedErr: true},
+	tt := []testCase{
+		{name: "Divide two positive numbers", a: 6, b: 2, c: []float64{}, want: 3, expectedErr: false},
+		{name: "Divide two negative numbers", a: -10, b: -5, want: 2, expectedErr: false},
+		{name: "Divide two negative fraction numbers", a: -10.5, b: -5, want: 2.1, expectedErr: false},
+		{name: "Divide one positive, one negative number", a: -10, b: 2.5, want: -4, expectedErr: false},
+		{name: "Divide 0 by a number", a: 0, b: 5, want: 0, expectedErr: false},
+		{name: "Divide by zero", a: 2.5, b: 0.5, c: []float64{2, 1.25, 1}, want: 2, expectedErr: false},
+
+		// Test cases with expected errors.
+		{name: "Divide by zero", a: 2, b: 0, want: 0, expectedErr: true},
+		{name: "Divide by zero", a: 2, b: 1, c: []float64{2, 0, 1}, want: 0, expectedErr: true},
 	}
 
 	for _, tc := range tt {
-		got, err := calculator.Multiply(tc.nums...)
+		t.Run(tc.name, func(t *testing.T) {
 
-		if err != nil && !tc.expectedErr {
-			t.Errorf("%s, Multiply(%v) should return an error", tc.name, tc.nums)
-		}
+			got, err := calculator.Divide(tc.a, tc.b, tc.c...)
 
-		if tc.want != got {
-			t.Errorf("%s, want %f, got %f", tc.name, tc.want, got)
-		}
+			if (err != nil) != tc.expectedErr {
+				t.Fatalf("%s; Divide(%f, %f, %v) should return error", tc.name, tc.a, tc.b, tc.c)
```

Comment:
But the reason for the failure might be that it returned an error when it wasn't supposed to! A good way to deal with this is to give a message like "unexpected error _status_: %v".